docgenhtml/plugins/org.polarsys.capella.docgen/src/org/polarsys/capella/docgen/util/StringUtil.java

```diff
@@ -465,12 +472,18 @@ private static String buildResourceRepresentationString(Resource resource, Strin
 			});
 		}
 
-		if (eObject instanceof DSemanticDiagram) {
-			stringBuilder.append(CapellaServices.getPathFromElement(((DSemanticDiagram) eObject).getTarget()));
-			stringBuilder.append("#");
-			stringBuilder.append(CapellaServices.getDiagramUid((DSemanticDiagram) eObject));
+		if (eObject != null) {
```

Comment:
The instanceof return false if the object is null. why did you add this null check?

Reply:
Because it may call `stringBuilder.append(CapellaServices.getPathFromElement(eObject));` on null

Reply:
l481